File: Preparation/HW5/Traffic/grader.c
#include "traffic.h"
#include <stdio.h>

static int N,P[1000000],S[1000000],D[1000000];

int main(){
   int i;
   scanf("%d",&N);
   for (i=0;i<N;i++) scanf("%d",&P[i]);
   for (i=0;i<N-1;i++) scanf("%d%d",&S[i],&D[i]);
   int r = LocateCentre(N,P,S,D);
   printf("%d\n",r);
   return 0;
}

File: Preparation/HW5/Traffic/grader.pas
program grader;
   uses traffic;
   var i,r,N: longint;
   var P,S,D: array [0..1000000] of longint;
begin
   read(N); 
   for i:=0 to N-1 do read(P[i]);
   for i:=0 to N-2 do read(S[i],D[i]);
   r := LocateCentre(N,P,S,D);
   writeln(r);
end.
